authors/apps/articles/views.py

```diff
@@ -61,13 +63,17 @@ def get_object(self, slug):
     def get(self, request, slug):
         article = self.get_object(slug)
         context = {"request": request}
-        if article:
-            serialized_data = self.serializer_class(article,
-                                                    context=context)
-            return Response(serialized_data.data, status=status.HTTP_200_OK)
-        return Response({
-            'errors': 'sorry article with that slug doesnot exist'
-        }, status=status.HTTP_404_NOT_FOUND)
+        if not article:
+            return Response({
+                'errors': 'sorry article with that slug doesnot exist'
+            }, status=status.HTTP_404_NOT_FOUND)
+        serialized_data = self.serializer_class(article,
+                                                context=context)
+        if not request.user.is_anonymous and\
```

Comment:
A few comments here would be helpful @codjoero

Reply:
@Karuhanga thanks for the suggestion, I went ahead and added some comments where I thought the code was abit complex.